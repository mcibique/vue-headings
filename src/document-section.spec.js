/* eslint-env node, jest */

import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import DocumentSection from "./document-section.vue";

describe("DocumentSection component", function () {
  function mountDocumentSection (props, mountOpts) {
    return mount(DocumentSection, {
      propsData: { ...props },
      ...mountOpts,
    });
  }

  it("should render properly", function () {
    let wrapper = mountDocumentSection();
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should render as the given tag", function () {
    let wrapper = mountDocumentSection({ tag: "main" });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should render the default slot", function () {
    let wrapper = mountDocumentSection({}, {
      scopedSlots: {
        default: "<span>Random text</span>",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
