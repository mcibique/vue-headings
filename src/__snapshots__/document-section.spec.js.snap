// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DocumentSection component > should render as the given tag 1`] = `<main />`;

exports[`DocumentSection component > should render properly 1`] = `<section />`;

exports[`DocumentSection component > should render the default slot 1`] = `
<section>
  <span>
    Random text
  </span>
</section>
`;
